src/main/java/com/zendesk/maxwell/replication/BinlogPosition.java

```diff
@@ -1,34 +1,64 @@
 package com.zendesk.maxwell.replication;
 
+import com.github.shyiko.mysql.binlog.GtidSet;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 import java.io.Serializable;
 import java.sql.Connection;
 import java.sql.ResultSet;
 import java.sql.SQLException;
 
 public class BinlogPosition implements Serializable {
+	static final Logger LOGGER = LoggerFactory.getLogger(BinlogPosition.class);
+
+	private static final String FILE_COLUMN = "File";
+	private static final String POSITION_COLUMN = "Position";
+	private static final String GTID_COLUMN = "Executed_Gtid_Set";
+
+	private final String gtidStr;
 	private final long offset;
 	private final String file;
 	private final Long heartbeat;
 
-	public BinlogPosition(long l, String file, Long heartbeat) {
+	public BinlogPosition(String gtidStr, long l, String file, Long heartbeat) {
+		this.gtidStr = gtidStr;
 		this.offset = l;
 		this.file = file;
 		this.heartbeat = heartbeat;
 	}
 
+	public BinlogPosition(long l, String file, Long heartbeat) {
+		this(null, l, file, heartbeat);
+	}
+
 	public BinlogPosition(long l, String file) {
-		this(l, file, null);
+		this(null, l, file, null);
 	}
 
 	public static BinlogPosition capture(Connection c) throws SQLException {
 		ResultSet rs;
 		rs = c.createStatement().executeQuery("SHOW MASTER STATUS");
 		rs.next();
-		return new BinlogPosition(rs.getInt("Position"), rs.getString("File"));
+		long l = rs.getInt(POSITION_COLUMN);
+		String file = rs.getString(FILE_COLUMN);
+		String gtidStr = null;
+		try {
+			gtidStr = rs.getString(GTID_COLUMN);
+		} catch (SQLException ex) {
+			// gtid column might not exist if gtid mode is not turned on
+			LOGGER.info("Unable to retrieve column " + GTID_COLUMN, ex);
```

Comment:
no need to log here, this is a very normal case if the server doesn't have gtid-support.